quickwit/quickwit-control-plane/src/debouncer.rs

```diff
@@ -91,25 +91,25 @@ impl Debouncer {
         previous_state
     }
 
-    fn emit_message<A, M>(&self, ctx: &ActorContext<A>)
+    fn emit_message<A, M>(&self, ctx: &ActorContext<A>, message: M)
     where
         A: Actor + Handler<M> + DeferableReplyHandler<M>,
-        M: Default + std::fmt::Debug + Send + Sync + 'static,
+        M: std::fmt::Debug + Send + Sync + 'static,
     {
-        let _ = ctx.mailbox().send_message_with_high_priority(M::default());
+        let _ = ctx.mailbox().send_message_with_high_priority(message);
     }
 
-    fn schedule_post_cooldown_callback<A, M>(&self, ctx: &ActorContext<A>)
+    fn schedule_post_cooldown_callback<A, M>(&self, ctx: &ActorContext<A>, message: M)
```

Comment:
I think there are a couple of ugly things here, but maybe this is ok.

The reason why I did not put `message` in the API is because this suppose a bit of a footgun.
If someone sends different messages, the message that will be shadowed through deboucning will be quite random.

Another thing I don't like is that `schedule_post_cooldown_callback` taking message makes no sense at all today. It is an artefact of the implementation.

Can you see if you can remove it by storing the message within the `DebouncerState::CooldownScheduled(Message)` maybe?

Reply:
I decided to go with a much simpler debouncing as we don't really care if we miss out on some pruning requests